src/citrine/ara/variables.py

```diff
@@ -101,3 +110,232 @@ def __init__(self, *,
+                 scope: str):
+        self.name = name
+        self.headers = headers
+        self.processTemplate = processTemplate
+        self.ingredientName = ingredientName
+        self.scope = scope
+
+
+class IngredientLabelByProcessAndName(Serializable['IngredientLabelByProcessAndName'], Variable):
+    """Define a boolean variable indicating whether a given label is applied to an in ingredient
+    that is associated with a process template and a name.
+    
+    For example, a label might be "solvent" for the variable "is the ethanol being used as a
+    solvent?".  Many such columns would then support the downstream analysis "get the volumetric
+    average density of the solvents".
+
+    Parameters
+    ---------
+    name: str
+        a short human-readable name to use when referencing the variable
+    headers: list[str]
+        sequence of column headers
+    processTemplate: LinkByUID
+        process template associated with this ingredient identifier
+    ingredientName: str
+        name of ingredient
+    label: str
+        label to test
+    """
+
+    name = properties.String('name')
+    headers = properties.List(properties.String, 'headers')
+    processTemplate = properties.Object(LinkByUID, 'processTemplate')
+    ingredientName = properties.String('ingredientName')
+    label = properties.String('label')
+    typ = properties.String('type', default="ing_label_by_process_and_name", deserializable=False)
+
+    def _attrs(self) -> List[str]:
+        return ["name", "headers", "processTemplate", "ingredientName", "label", "typ"]
+
+    def __init__(self, *,
+                 name: str,
+                 headers: List[str],
+                 processTemplate: LinkByUID,
+                 ingredientName: str,
+                 label: str):
+        self.name = name
+        self.headers = headers
+        self.processTemplate = processTemplate
+        self.ingredientName = ingredientName
+        self.label = label
+
+
+class IngredientQuantityByProcessAndName(Serializable['IngredientQuantityByProcessAndName'], Variable):
+    """Get the dimension of the quantity in an ingredient that is associated with a process
+    template and a name.
+    
+    Parameters
+    ---------
+    name: str
+        a short human-readable name to use when referencing the variable
+    headers: list[str]
+        sequence of column headers
+    processTemplate: LinkByUID
+        process template associated with this ingredient identifier
+    ingredientName: str
+        name of ingredient
+    quantity: IngredientQuantity
+        dimension of the ingredient quantity (e.g. absolute, number fraction...)
+    """
+
+    name = properties.String('name')
+    headers = properties.List(properties.String, 'headers')
+    processTemplate = properties.Object(LinkByUID, 'processTemplate')
+    ingredientName = properties.String('ingredientName')
+    quantityDimension = properties.Integer('quantityDimension')
```

Comment:
Based on the platform-backend definition, I think that this variable will be serialized with the key "quantity". I do see the logic for changing it to `quantityDimension` for the purposes of this user-facing client. That can be done here. For example, you'll notice that the line below is `typ = properties.String('type')`. That says that it should be serialized as "type" but deserialized as "typ". You can do the same thing for quantity/quantityDimensions, and then create an Enum object and in the `__init__` method apply that Enum object to the value of `quantityDimension` to ensure that corresponds to one of the four expected values.

Reply:
I went ahead and added an Enumerated property type.